src/containers/flows/BuyNow/index.tsx

```diff
@@ -95,22 +103,28 @@ const BuyNowContainer = ({
 		setCurrentStep(Step.WaitingForWalletConfirmation);
 		try {
 			if (!sdk || !sdk.zauction) {
-				throw Error('Failed to retrieve zAuction instance');
+				throw Error(ERRORS.FAILED_TO_CHECK_ZAUCTION);
 			}
 			const tx = await sdk.zauction.buyNow(
 				{ amount: data!.buyNowPrice.toString(), tokenId: domainId },
 				library.getSigner(),
 			);
-			setCurrentStep(Step.Buying);
-			await tx.wait();
-			addNotification(`You have successfully purchased ${data?.title}`);
+			try {
+				setCurrentStep(Step.Buying);
+				await tx.wait();
+			} catch (e) {
+				setCurrentStep(Step.Details);
+				setError(ERRORS.TRANSACTION);
+			}
```

Comment:
Thanks for that, does the change here make sense to pass ERRORS.TRANSACTION? My understanding is this would throw the transaction failed message to the above catch, resulting in the error printing in the console, and the user would see 'Unexpected problem .....' (generic message), because I'm using the getDisplayErrorMessage function in the above catch

```setError(getDisplayErrorMessage(e.message));```